pkg/sync/sync_context.go

```diff
@@ -557,6 +569,15 @@ func (sc *syncContext) Sync() {
 		// delete all completed hooks which have appropriate delete policy
 		sc.deleteHooks(hooksPendingDeletionSuccessful)
 		sc.setOperationPhase(common.OperationSucceeded, "successfully synced (no more tasks)")
+
+		// Invalidate cache after successful sync
+		if sc.cacheInvalidationCallback != nil {
+			modifiedResources := make([]kubeutil.ResourceKey, 0, len(sc.syncRes))
+			for _, result := range sc.syncRes {
+				modifiedResources = append(modifiedResources, result.ResourceKey)
+			}
+			sc.cacheInvalidationCallback(modifiedResources)
+		}
```

Comment:
Same comment applies:

Why would we need to invalidate the cache on every sync if the cluster cache is based on resource watches? I feel that the root problem is elsewhere. WDYT?